core/src/test/java/org/elasticsearch/cluster/metadata/IndexCreationTaskTests.java

```diff
@@ -447,4 +447,52 @@ private void setupIndicesService() throws Exception {
 
         when(indicesService.createIndex(anyObject(), anyObject())).thenReturn(service);
     }
+
+    private <K> Matcher<ImmutableOpenMap<K, ?>> containsKey(final K key) {
```

Comment:
You can put these matchers in test-framework under `org.elasticsearch.test.hamcrest`.